lib/factorial.rb

```diff
@@ -1,6 +1,16 @@
 # Computes factorial of the input number and returns it
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the number
+# Space complexity: O(1)
 def factorial(number)
-  raise NotImplementedError
+  return 1 if number == 0
+  return 1 if number == 1
+  raise ArgumentError if number == nil
```

Comment:
If someone put in factorial("2") you'd get an exception. Consider checks such as `if number.is_a?(Integer)`